src/RevocationPolicy.ts

```diff
@@ -0,0 +1,22 @@
+import {
+    Struct,
+    Field
+} from 'o1js';
+
+/** When a token is issued, issuer and holder should agree on who can revoke the token */
+class RevocationPolicy extends Struct ({
+    type: Field,
+}) {
+    public static types = {
+        // the issuer has the right to revoke tokens
+        issuerOnly: Field(0),
+        // only the holder of the token can revoke it
+        holderOnly: Field(1),
+        // a token can only be revoked if issuer and holder agree
+        both: Field(2),
+        // tokens are indestructible
+        neither: Field(3),
+    };
+}
+
+export { RevocationPolicy };
```

Comment:
There's potentially two ways to use the `RevocationPolicy`
- Instead of making this a `Struct` we can make it another `MerkleMap` who's root is committed to the contract at deployment time and use a `ZkProgram` to attest to the `RevocationPolicy` of the SBT. 

- Alternatively, the `SmartContract` could have a state that defines the `RevociationPolicy` (e.g. a hash or field) and a method check if the input `Struct` argument satisfies those constraints.